### PR TITLE
Add quota checking for VMReconfigure tests.

### PR DESCRIPTION
**Pull request by billfitzgerald0120**
Updated tests to include quota checking for VMReconfigure.

### CODE REVIEW

**Comment by billfitzgerald0120** [comment]
@tinaafitz 
Please Review

**Comment by billfitzgerald0120** [comment]
@miq-bot add_label bug, fine/yes

**Comment by billfitzgerald0120** [comment]
@tinaafitz 
Please Review

**Comment by billfitzgerald0120** [comment]
modified delete disk to handle 0 bytes
@tinaafitz 
Please review

**Comment by tinaafitz** [comment]
@billfitzgerald0120 Looks good.
@mkanoor Please review.

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb [hunk 1]
```diff
@@ -120,8 +146,35 @@ def vmdb_object(model, id)
...
+    # Adding/removing disks only supported for VMware
+    if args_hash[:resource].options[:disk_add]
+      args_hash[:resource].options[:disk_add].each do |disk|
+        $evm.log(:info, "Adding a disk:  #{disk.inspect}")
+        args_hash[:prov_value] += disk['disk_size_in_mb'].to_i.megabytes
+      end
+    end
+    if args_hash[:resource].options[:disk_remove]
+      args_hash[:resource].options[:disk_remove].each do |disk|
+        $evm.log(:info, "Removing disk: #{disk.inspect}")
```
Comment (mkanoor) [inline content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb]: @billfitzgerald0120 
Can we move this line to line 163 and change the message to  Disk Info

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb [hunk 2]
```diff
@@ -120,8 +146,35 @@ def vmdb_object(model, id)
...
+        disk_num = disk[:disk_name].match(/_(\d).vmdk/)
+        next unless disk_num
+        disk_n_number = "disk_#{disk_num[1].succ}_size"
+        disk_n_size   = @vm.send(disk_n_number.to_s)
+        $evm.log(:info, "Removing disk size: #{disk_n_size.to_s(:human_size)}") if disk_n_size
```
Comment (mkanoor) [inline content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb]: @billfitzgerald0120 
can use use next unless disk_n_size and change the log message to 
$evm.log(:info, "Disk size:  #{disk_n_size.to_s(:human_size)}")

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb [hunk 3]
```diff
@@ -265,4 +318,13 @@ def error(type)
   exit MIQ_OK
 end
 
+@reconfigure_request = @miq_request.type == "VmReconfigureRequest"
+if @reconfigure_request
+  @check_quota = false # default, unless additional quota is requested
+  vm_id = @miq_request.options[:src_ids]
+  @vm = $evm.vmdb(:vm).find_by(:id => vm_id)
```
Comment (mkanoor) [inline content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb]: @billfitzgerald0120 
Can we raise an error if the vm is not found.

Comment (billfitzgerald0120) [inline content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb]: @mkanoor 
Made requested changes.

**Comment by simaishi** [comment]
@billfitzgerald0120 This is causing Travis failure in Fine branch (https://travis-ci.org/ManageIQ/manageiq-content/builds/240035211) Can you please take a look?

```
NoMethodError:
       undefined method `vmware_reconfigure_model' for #<RSpec::ExampleGroups::QuotaValidation_2::VmReconfigQuotaCalculateRequest:0x00000004f90d90>
```

**Comment by gmcculloug** [comment]
Looks like we also need to back port https://github.com/ManageIQ/manageiq/pull/14756.

**Comment by billfitzgerald0120** [comment]
Yes, This is required.
https://github.com/ManageIQ/manageiq/pull/14756

**Comment by simaishi** [comment]
Thanks - backported https://github.com/ManageIQ/manageiq/pull/14756, will re-run Travis.

**Pull request merged**